pkg/odo/cli/plugins/plugin_handler_test.go

```diff
@@ -0,0 +1,66 @@
+package plugins
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"runtime"
+	"testing"
+)
+
+var sampleScript = []byte(`
+#!/bin/sh
+echo 'hello'
+`)
+
+func TestDefaultHandlerLookup(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip()
+	}
+	tempDir, cleanup := makeTempDir(t)
```

Comment:
Ideally unit tests shouldn’t interact with external entities such as a filesystem which are subject to change. Can you please use the mocked filesystem we have inplace?

Reply:
Right, @girishramnani Can you please add some reference where we have already done it. Reference would really be helpful.

Reply:
- https://github.com/openshift/odo/blob/49eae2d4dd57d09c6180e7d6cf70df5edd70cd4a/pkg/config/config_test.go#L446
- https://github.com/openshift/odo/blob/5e9c163aac521fae608f15486bd3eb0e73c7d1c6/pkg/debug/info_test.go#L38

Reply:
- https://github.com/openshift/odo/blob/a93f4dae81282475a6e58bf025411f516329afcb/pkg/devfile/parser/content_test.go#L10

Reply:
This looks like I'd need to reimplement https://golang.org/pkg/os/exec/#LookPath on top of afero.

In other words, replace https://github.com/golang/go/blob/master/src/os/exec/lp_unix.go#L34 with a version that worked on top of afero, purely for test purposes.

I'm not sure of the value of this?

Reply:
Then maybe consider writing an integration test?